src/Evolution/DgSubcell/Matrices.cpp

```diff
@@ -20,11 +20,76 @@
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Numeric.hpp"  // IWYU pragma: keep
+#include "Utilities/StaticCache.hpp"
 
 namespace evolution::dg::subcell::fd {
+template <size_t Dim>
+const Matrix& projection_matrix(const Mesh<Dim>& dg_mesh,
+                                const Index<Dim>& subcell_extents) {
```

Comment:
Since this function returns a 1D matrix, how about you take a `Mesh<1> dg_mesh` and a `size_t subcell_extents` here? This is how it's done in `NumericalAlgorithms/Spectral/Projection.hpp`. There's also an overload that returns the 3D matrix (as a `std::array<std::reference_wrapper<const Matrix>, Dim>`), taking a `Mesh<Dim>` and `Index<Dim>`.